src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -704,11 +703,28 @@ public String candleData(JSONObject params) {
 		}
 	}
 
+	/**
+	 * Get Market Data.
+	 *
+	 * @param params is market data params.
+	 * @return returns the details of market data.
+	 */
+	public JSONObject marketData(JSONObject params) {
+		try{
+			String url = routes.get("api.market.data");
+			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
+			return response.getJSONObject("data");
+		}catch (Exception | SmartAPIException e) {
+			System.out.println(e.getMessage());

```

Comment:
Why are you using System.out.println(e.getMessage());?

Reply:
I Have taken references from other methods, In the catch block the error message is printed.
I can use logger instead of System.out.println and handle null but the code will differ from other methods.

some referred methods 
public String candleData(JSONObject params) {
		try {
			String url = routes.get("api.candle.data");
			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
			System.out.println(response);
			return response.getString("data");
		} catch (Exception | SmartAPIException e) {
			System.out.println(e.getMessage());
			return null;
		}
	}
	
	public JSONObject gttRuleDetails(Integer id) {
		try {

			JSONObject params = new JSONObject();
			params.put("id", id);

			String url = routes.get("api.gtt.details");
			JSONObject response = smartAPIRequestHandler.postRequest(this.apiKey, url, params, accessToken);
			System.out.println(response);

			return response.getJSONObject("data");
		} catch (Exception | SmartAPIException e) {
			System.out.println(e.getMessage());
			return null;
		}

	}